diff --git a/node_modules/react-native-cloud-storage/ios/RCTCloudStorageCloudKit.mm b/node_modules/react-native-cloud-storage/ios/RCTCloudStorageCloudKit.mm
index 3098fa4..a8f4c21 100644
--- a/node_modules/react-native-cloud-storage/ios/RCTCloudStorageCloudKit.mm
+++ b/node_modules/react-native-cloud-storage/ios/RCTCloudStorageCloudKit.mm
@@ -61,8 +61,19 @@
   [self emitCloudAvailabilityChanged];
 }
 
+- (void)emitOnCloudAvailabilityChanged:(NSDictionary *)value
+{
+  if (!static_cast<bool>(_eventEmitterCallback)) {
+    return;
+  }
+  [super emitOnCloudAvailabilityChanged:value];
+}
+
 - (void)emitCloudAvailabilityChanged
 {
+  if (!static_cast<bool>(_eventEmitterCallback)) {
+    return;
+  }
   BOOL isCloudAvailable = [NSFileManager defaultManager].ubiquityIdentityToken != nil;
   [self emitOnCloudAvailabilityChanged:@{ @"available" : @(isCloudAvailable) }];
 }
